azkaban-common/src/main/java/azkaban/executor/container/watch/AzPodStatusExtractor.java

```diff
@@ -0,0 +1,527 @@
+   * conditions [0 - 4] present in them. Optional.empty() denotes a missing condition in the event.
+   * @see
+   * <a href="https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/">Pod Lifecycle</a>
+   */
+  private void extractConditions() {
+    requireNonNull(podConditions, "pod status conditions must not be null");
+    Map<String, V1PodCondition> conditionMap = new HashMap<>();
+    this.podConditions.stream().forEach(
+        condition ->
+            conditionMap.put(condition.getType(), condition));
+  this.scheduledCondition =
+      Optional.ofNullable(conditionMap.remove(PodCondition.PodScheduled.name()));
+  this.containersReadyCondition =
+      Optional.ofNullable(conditionMap.remove(PodCondition.ContainersReady.name()));
+  this.initializedCondition =
+      Optional.ofNullable(conditionMap.remove(PodCondition.Initialized.name()));
+  this.readyCondition = Optional.ofNullable(conditionMap.remove(PodCondition.Ready.name()));
+
+  conditionMap.keySet().stream().forEach(type -> logger.warn("Unexpected condition of type: " + type));
+  }
+
+  /**
+   * For each of the Conditions present in the the event, extract the corresponding status.
+   * Optional.empty() denotes the corresponding condition was not present.
+   */
+  private void extractConditionStatuses() {
+    this.scheduledCondition.ifPresent(cond -> {
+      requireNonNull(cond.getStatus());
+      this.scheduledConditionStatus = Optional.of(PodConditionStatus.valueOf(cond.getStatus()));
+    });
+    this.containersReadyCondition.ifPresent(cond -> {
+      requireNonNull(cond.getStatus());
+      this.containersReadyConditionStatus = Optional
+          .of(PodConditionStatus.valueOf(cond.getStatus()));
+    });
+    this.initializedCondition.ifPresent(cond -> {
+      requireNonNull(cond.getStatus());
+      this.initializedConditionStatus =
+          Optional.of(PodConditionStatus.valueOf(cond.getStatus()));
+    });
+    this.readyCondition.ifPresent(cond -> {
+      requireNonNull(cond.getStatus());
+      this.readyConditionStatus = Optional.of(PodConditionStatus.valueOf(cond.getStatus()));
+    });
+  }
+
+  /**
+   * Extract the Pod Phase from the event.
+   * @see
+   * <a href="https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/">Pod Lifecycle</a>
+   */
+  private void extractPhase() {
+    requireNonNull(this.v1PodStatus.getPhase(), "pod status phase must not be null");
+    // This will throw an IllegalArgumentException in case of an unexpected phase name.
+    this.podPhase = PodPhase.valueOf(v1PodStatus.getPhase());
+  }
+
+  /**
+   * Evaluate if event can be classified as status {@link AzPodStatus.AZ_POD_REQUESTED}
+   * @return true if the classification succeeds, false otherwise.
+   */
+  private boolean checkForAzPodRequested() {
+    // Scheduled conditions should either not be present or be false
+    if (scheduledConditionStatus.isPresent() &&
```

Comment:
nitpick : please use "this" keyword when using member variables.

Reply:
Done.